runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -1124,6 +1124,20 @@ TR::SymbolValidationManager::addJ2IThunkFromMethodRecord(void *thunk, TR_OpaqueM
    appendNewRecord(thunk, record);
    }
 
+bool
+TR::SymbolValidationManager::addIsClassVisibleRecord(TR_OpaqueClassBlock *sourceClass, TR_OpaqueClassBlock *destClass, bool isVisible)
+   {
+   SVM_ASSERT_ALREADY_VALIDATED(this, sourceClass);
+   SVM_ASSERT_ALREADY_VALIDATED(this, destClass);
+
+   if (sourceClass == destClass
+      || _fej9->isJavaLangObject(sourceClass)
+      || _fej9->isJavaLangObject(destClass))
```

Comment:
I would add a comment here for these two, similar to the InstantOf validation. The reason we can ignore if the `destClass` is j/l/Object is because Object is always visible. The reason we can ignore if the `sourceClass` is j/l/Object is because nothing is visible to Object., though I'm not 100% sure about this point. @jdmpapin could you confirm?

Reply:
If we're going to skip generating a record, we have to check that `isVisible` has the value that we expect. For example, if we think that `Object` should be visible, but `!isVisible`, then the record is still necessary.

I almost got confused here and talked about whether `sourceClass` is able to find `destClass` by name, since that seems highly relevant to "visibility." But it looks like the underlying VM function `checkVisibility()` is purely about access control (JVMS section 5.4.4). Given that, I agree that `Object` is always visible. However, I don't think there are any special rules about what's visible to `Object`.

I'm not sure that we need to, but we could eliminate more records (including some with `isJavaLangObject(sourceClass)`) just based on the fully qualified name and visibility, which are already determined by the class chains. For example, I believe no record is required if one of the following are true:
- `isVisible`, and `destClass` is public in Java 8; or
- `!isVisible`, the classes are in differently named packages, and `destClass` is not `public`.

For Java 11+, modules really complicate the visibility tests, but there are still probably things that could be done to eliminate records involving classes in java.base

Reply:
@dylanjtuttle I think for the purpose of this PR, it's best to remove 
```
      || _fej9->isJavaLangObject(sourceClass)
```

Reply:
Fixed in [b86ed1b](https://github.com/eclipse-openj9/openj9/pull/16000/commits/b86ed1bc3fc63a42808cf832d45cd5f5c6278d8e) and added the suggested comment in [5ef2446](https://github.com/eclipse-openj9/openj9/pull/16000/commits/5ef24464fccff29abf776a3ea6700c2b1f7fd4d4).

Reply:
By the way, I retract this statement:
> If we're going to skip generating a record, we have to check that `isVisible` has the value that we expect.

It came from my line of thinking about a sneaky class loader hiding `Object`, which turned out to be irrelevant. In the case where `destClass` is `Object`, `isVisible` should be guaranteed. It might make a good assertion, but checking it isn't necessarily required.

For posterity, it's similar in the examples I gave of other cases where the record could be eliminated. That is:
- if `destClass` is public in Java 8, then `isVisible` should be guaranteed; and
- if the classes are in differently named packages, and `destClass` is not public, `!isVisible` should be guaranteed.